tests/handover_net.rs

```diff
@@ -113,33 +115,33 @@ impl Net {
             ballot: self.gen_ballot(recursion, faulty_nodes, rng),
         };
 
-        let mut signed_vote = faulty_node.sign_vote(vote).unwrap();
-        let node_to_impersonate = self.procs.iter().choose(rng).unwrap().public_key_share();
-        signed_vote.voter = node_to_impersonate;
-        signed_vote
+        SignedVote {
+            voter: self.pick_id(rng),
```

Comment:
Just realised this might just well randomly choose itself and be a perfectly valid signed vote

Reply:
That's a good point, lets rename `gen_faulty_*` to `gen_random_*`. Next PR

Reply:
created #17 to track that